devito/ir/support/basic.py

```diff
@@ -588,9 +588,6 @@ def is_indep(self, dim=None):
                 return False
             elif dim is None:
                 return self.distance == 0
-            elif self.source.is_local and self.sink.is_local:
```

Comment:
fixed somewhere else?

Reply:
not sure when, possibly months ago through the various refinements to data dependence analysis. This `elif` was probably added a loong time ago when the analysis wasn't as sophisticated as it is now. So I realised that it now was actually causing false positives. So I tried to drop it and everything (all tests) were still working

Reply:
and this is a good thing, especially here in this file -- the lighter these routines are, the better, and not just because of speed, but also because despite they are fairly neat, the complexity they hide is quite scary